docs/deployment-patterns/deployment-as-code.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Deployment as code
+description: You can define your entire Octopus configuration as code and track it all through source control. This page describes the different options available right now, and our plans to make this even better in the future.
+position: 4
+---
+
+You are probably reading this page because you want to manage your **deployment as code**. You want to define your deployment process, scripts, and variables in code, perhaps colocating this with the application source code in your git repository. This page describes the different options available in Octopus right now, and our plans to make this even better in the future.
```

Comment:
I would use the phrase `configuration as code` rather than `deployment as code`, but that's probable personal preference. It probably depends on the expectations you want to set.

Reply:
At first blush I agree, but thinking about it a bit more I think deployments hits the target quicker, using **configuration as code** might lead to more questions, where **deployment as code** gets people there straight away, I think :)

Reply:
I wrestled with similar concerns. I wanted to differentiate between the variety of ways you can "as code" things in Octopus. At the moment we offer:

1. `Scripts as code` (crazy I know, scripts are code, but this refers to the script being stored as code in your source control system)
2. `Project as code` (configure your whole Octopus project as code)
3. `Configuration as code` (configure your entire Octopus via code)

I can imagine a future where we offer something like `Process as code` or `Release as code`... mainly based on my rough experience coming up with "the simplest thing that could possibly work" when dealing with branches.

I'll leave it as is for now.